plugins/filter/replace_keys.py

```diff
@@ -0,0 +1,189 @@
+
+  # 4) Replace keys that match any regex of the attributes before.
+  t:
+    - {before: "^.*0_x.*$", after: a0}
+    - {before: "^.*1_x.*$", after: a1}
+  r: "{{ l | community.general.replace_keys(target=t, matching_parameter='regex') }}"
+
+  # The results of above examples 1-4 are all the same.
+  r:
+    - {a0: A0, a1: B0, k2_x2: [C0], k3_x3: foo}
+    - {a0: A1, a1: B1, k2_x2: [C1], k3_x3: bar}
+
+  # 5) If more keys match the same attribute before the last one will be used.
+  t:
+    - {before: "^.*_x.*$", after: X}
+  r: "{{ l | community.general.replace_keys(target=t, matching_parameter='regex') }}"
+
+  # gives
+
+  r:
+    - X: foo
+    - X: bar
+
+  # 6) If there are items with equal attribute before the last one will be used.
+  t:
+    - {before: "^.*_x.*$", after: X}
+    - {before: "^.*_x.*$", after: Y}
+  r: "{{ l | community.general.replace_keys(target=t, matching_parameter='regex') }}"
+
+  # gives
+
+  r:
+    - Y: foo
+    - Y: bar
+
+  # 7) If there are more matches for a key the first one will be used.
+  l:
+    - {aaa1: A, bbb1: B, ccc1: C}
+    - {aaa2: D, bbb2: E, ccc2: F}
+  t:
+    - {before: a, after: X}
+    - {before: aa, after: Y}
+  r: "{{ l | community.general.replace_keys(target=t, matching_parameter='starts_with') }}"
+
+  # gives
+
+  r:
+    - {X: A, bbb1: B, ccc1: C}
+    - {X: D, bbb2: E, ccc2: F}
+'''
+
+RETURN = '''
+  _value:
+    description: The list of dictionaries with replaced keys.
+    type: list
+    elements: dictionary
+'''
+
+from ansible_collections.community.general.plugins.plugin_utils.keys_filter import (
+    _keys_filter_params,
+    _keys_filter_target_dict)
+
+
+def replace_keys(data, target=None, matching_parameter='equal'):
+    """replace specific keys in a list of dictionaries"""
+
+    # test parameters
+    _keys_filter_params(data, target, matching_parameter)
+    # test and transform target
+    td = _keys_filter_target_dict(target, matching_parameter)
+
+    before = list(td.keys())
+    index = 0  # If there are multiple matches take the first one.
+
+    if matching_parameter == 'starts_with':
+        def match_key(key, t):
+            return key.startswith(t)
+    elif matching_parameter == 'ends_with':
+        def match_key(key, t):
+            return key.endswith(t)
+    elif matching_parameter == 'regex':
+        def match_key(key, t):
+            return t.match(key)
+
+    # If matching_parameter is 'equal' there may be a single match only.
+    if matching_parameter == 'equal':
+        def replace_key(key):
+            if key in before:
+                return td[key]
+            else:
+                return key
+    # Otherwise, there may be multiple matches. In this case, use index.
+    else:
+        def replace_key(key):
+            rl = []
+            for t in before:
+                if match_key(key, t):
+                    rl.append(td[t])
+            if rl:
+                return rl[index]
+            else:
+                return key
```

Comment:
Why not remove `index` and simplify this to:
```suggestion
            for t in before:
                if match_key(key, t):
                    return td[t]
            return key
```

Reply:
Done. Implemented in the last PR.